microsoft/testsuites/dpdk/dpdkutil.py

```diff
@@ -1085,4 +1099,80 @@ def create_l3fwd_rules_files(
     )
 
 
-DPDK_VERSION_TO_RDMA_CORE_MAP = {"20.11": "46.1", "21.11": ""}
+# Device name match for LSPCI
+class NicType(Enum):
+    CX3 = "ConnectX-3"
+    CX4 = "ConnectX-4"
+    CX5 = "ConnectX-5"
+    MANA = "Device 00ba"
+
+
+# Short name for nic types
+NIC_SHORT_NAMES = {
+    NicType.CX3: "cx3",
+    NicType.CX4: "cx4",
+    NicType.CX5: "cx5",
+    NicType.MANA: "mana",
+}
+
+
+# Get the short name for the type of nic on a node
+def get_node_nic_short_name(node: Node) -> str:
+    devices = node.tools[Lspci].get_devices_by_type(DEVICE_TYPE_SRIOV)
+    if node.nics.is_mana_device_present():
+        return NIC_SHORT_NAMES[NicType.MANA]
+    for nic_name in [NicType.CX3, NicType.CX4, NicType.CX5]:
+        if any([str(nic_name) in x.device_id for x in devices]):
+            return NIC_SHORT_NAMES[nic_name]
+    # We assert much earlier to enforce that SRIOV is enabled,
+    # so we should never hit this unless someone is testing a new platform.
+    # Instead of asserting, just log that the short name was not found.
+    known_nic_types = ",".join(
+        map(str, [NicType.CX3, NicType.CX4, NicType.CX5, NicType.MANA])
+    )
+    found_nic_types = ",".join(map(str, [x.device_id for x in devices]))
+    node.log.debug(
+        "Unknown NIC hardware was detected during DPDK test case. "
+        f"Expected one of: {known_nic_types}. Found {found_nic_types}. "
+    )
+    # this is just a function for annotating a result, so don't assert
+    # if there's
+    return ""
```

Comment:
It can returns the `found_nic_types`. it's easier to troubleshoot.

Reply:
I love this idea